sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -1832,16 +1833,28 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
   }
 
   test("SPARK-33468: ParseUrl in ANSI mode should fail if input string is not a valid url") {
+    val url = "https://a.b.c/index.php?params1=a|b&params2=x"
     withSQLConf(SQLConf.ANSI_ENABLED.key -> "true") {
       val msg = intercept[IllegalArgumentException] {
         evaluateWithoutCodegen(
-          ParseUrl(Seq("https://a.b.c/index.php?params1=a|b&params2=x", "HOST")))
+          ParseUrl(Seq(url, "HOST")))
       }.getMessage
       assert(msg.contains("Find an invalid url string"))
     }
-    withSQLConf(SQLConf.ANSI_ENABLED.key -> "false") {
+    withSQLConf(SQLConf.ANSI_ENABLED.key -> "true") {
+            checkError(exception = intercept[SparkIllegalArgumentException]{
+              evaluateWithoutCodegen(
+             ParseUrl(Seq(url, "HOST")))
```

Comment:
Could you trigger the error from user code, please. For example `sql("select parse_url(...)")`